### PR TITLE
Add new source

### PR DESCRIPTION
**Pull request by randrii**
- one more source
- @jsonalias added
- renaming & refactoring

### CODE REVIEW

File: src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    private final ObjectMapper mapper;
+
+    public ExchangeRatesApiConnectorService(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public ExchangeResponse retrieveRates(String userBaseCurrency) {
+
+        try {
+            CVurl cVurl = new CVurl();
```
Comment (jarc0der) [inline src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: Again, why you didn't provide dependency through the constructor?
Every call of retrieveRates create a new CVurl client...

File: src/main/java/com/rybka/CurrencyAPIConstants.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package com.rybka;
 
 public final class CurrencyAPIConstants {
-    public static final String API_URL = "https://prime.exchangerate-api.com/v5/%s/latest/%s";
-    public static final String API_KEY = "4e0bbdc44fcdf05612fa0882";
+    public static final String EXCHANGE_RATE_API_URL = "https://prime.exchangerate-api.com/v5/%s/latest/%s";
+    public static final String EXCHANGE_RATE_API_KEY = "4e0bbdc44fcdf05612fa0882";
+    public static final String EXCHANGE_RATE__API_URL = "https://api.exchangeratesapi.io/latest?base=%s";
```
Comment (jarc0der) [inline src/main/java/com/rybka/CurrencyAPIConstants.java]: __ ? What convention have you followed? 

PRIME_EXCHANGE_RATE_API_URL = "https://prime.exchangerate-api.com/v5/%s/latest/%s";
PRIME_EXCHANGE_RATE_API_KEY = "4e0bbdc44fcdf05612fa0882";
EXCHANGE_RATE_API_URL  = "https://api.exchangeratesapi.io/latest?base=%s";

File: src/main/java/com/rybka/service/ExchangeRate_ApiConnectorService.java [hunk 3]
```diff
@@ -12,19 +12,19 @@
 import java.net.http.HttpResponse;
 
 @Log4j
-public class ApiConnectorService {
+public class ExchangeRate_ApiConnectorService implements BaseApiConnectorService {
```
Comment (jarc0der) [inline src/main/java/com/rybka/service/ExchangeRate_ApiConnectorService.java]: Rename this class names...

File: src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            var response = cVurl.get(String.format(CurrencyAPIConstants.EXCHANGE_RATE__API_URL, userBaseCurrency))
+                    .asString()
+                    .orElseThrow(RuntimeException::new).getBody();
+            return mapper.readValue(response, ExchangeResponse.class);
```
Comment (jarc0der) [inline src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: CVurl can convert to object itself, try to look at the docs. 
Get rid of mapper here.

File: src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        } catch (JsonProcessingException exception) {
+            log.error("Error while calling Currency API service 2. Reason: " + exception.getMessage());
+            throw new CurrencyAPICallException("Wrong parameters in Currency API 2 request! Reason: " +
```
Comment (jarc0der) [inline src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: What does Currency API 2 means? How can I know what is 1 and 2? 
Try to come up with names for APIs and follow them in the code.

Comment (jarc0der) [inline src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: What `parameters` we talking about in this logs?

File: src/main/java/com/rybka/model/ExchangeResponse.java [hunk 6]
```diff
@@ -25,6 +26,6 @@
     @JsonProperty("base")
     private String base;
 
-    @JsonProperty("conversion_rates")
+    @JsonAlias({"conversion_rates", "rates"})
```
Comment (jarc0der) [inline src/main/java/com/rybka/model/ExchangeResponse.java:29]: Good choice. Did you consider to create separate DTO object?

File: src/main/java/com/rybka/Application.java [hunk 7]
```diff
@@ -2,21 +2,29 @@
...
+import com.rybka.service.ExchangeRatesApiConnectorService;
 import com.rybka.view.ExchangeView;
 
 import java.net.http.HttpClient;
 import java.util.Scanner;
 
 public class Application {
     public static void main(String[] args) {
+
+//        BaseApiConnectorService apiConnectorService = new ExchangeRate_ApiConnectorService(
```
Comment (jarc0der) [inline src/main/java/com/rybka/Application.java]: Comments?

File: src/main/java/com/rybka/service/BaseApiConnectorService.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.rybka.service;
+
+import com.rybka.model.ExchangeResponse;
+
+public interface BaseApiConnectorService {
```
Comment (jarc0der) [inline src/main/java/com/rybka/service/BaseApiConnectorService.java]: Rename to BaseCurrencyExchangeConnector.

File: src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java (hunk 1 above)
Comment (randrii) [reply src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: fixed

File: src/main/java/com/rybka/CurrencyAPIConstants.java (hunk 2 above)
Comment (randrii) [reply src/main/java/com/rybka/CurrencyAPIConstants.java]: fixed

File: src/main/java/com/rybka/service/ExchangeRate_ApiConnectorService.java (hunk 3 above)
Comment (randrii) [reply src/main/java/com/rybka/service/ExchangeRate_ApiConnectorService.java]: fixed

File: src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java (hunk 4 above)
Comment (randrii) [reply src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: fixed

File: src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java (hunk 5 above)
Comment (randrii) [reply src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: Fixed

Comment (randrii) [reply src/main/java/com/rybka/service/ExchangeRatesApiConnectorService.java]: fixed

File: src/main/java/com/rybka/Application.java (hunk 7 above)
Comment (randrii) [reply src/main/java/com/rybka/Application.java]: fixed

File: src/main/java/com/rybka/service/BaseApiConnectorService.java (hunk 8 above)
Comment (randrii) [reply src/main/java/com/rybka/service/BaseApiConnectorService.java]: fixed

File: src/main/java/com/rybka/model/ExchangeResponse.java (hunk 6 above)
Comment (randrii) [reply src/main/java/com/rybka/model/ExchangeResponse.java:29]: Thanks. Yes, I did, but I wanted to use JSON feature.

File: src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    private final CVurl cVurl;
+
+    public ExchangeRateBaseCurrencyExchangeConnector(CVurl cVurl) {
+        this.cVurl = cVurl;
+    }
+
+    public ExchangeResponse retrieveRates(String userBaseCurrency) {
+        return cVurl
+                .get(String.format(CurrencyAPIConstants.EXCHANGE_RATE_API_URL, userBaseCurrency))
+                .asObject(ExchangeResponse.class);
```
Comment (jarc0der) [inline src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java]: Looks much better. 
One more point, we need to throw our custom exception it case of request failure.

File: src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
+package com.rybka.service;
+
+import com.rybka.CurrencyAPIConstants;
+import com.rybka.model.ExchangeResponse;
+import coresearch.cvurl.io.request.CVurl;
+import lombok.extern.log4j.Log4j;
+
+@Log4j
+public class ExchangeRateBaseCurrencyExchangeConnector implements BaseCurrencyExchangeConnector {
```
Comment (jarc0der) [inline src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java]: I'd simplify this names to 
ExchangeRateConnector implements BaseCurrencyExchangeConnector
PrimeExchangeRateConnector implements BaseCurrencyExchangeConnector

File: src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java (hunk 9 above)
Comment (randrii) [reply src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java]: fixed

File: src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java (hunk 10 above)
Comment (randrii) [reply src/main/java/com/rybka/service/ExchangeRateBaseCurrencyExchangeConnector.java]: fixed

**Pull request merged**